### PR TITLE
Feat: Add thread-safe UserAttributes in read/write operations.

### PR DESCRIPTION
**Pull request by mnoman09**
## Summary
- Copying UserAttributes to make it thread safe.
- Made All fields in OptimizelyUserContext private and added getter methods

## Test plan
Updated all Unit tests

### CODE REVIEW

File: OptimizelySDK/OptimizelyUserContext.cs [hunk 1]
```diff
@@ -46,6 +46,21 @@ public OptimizelyUserContext(Optimizely optimizely, string userId, UserAttribute
...
+        }
+
+        public string GetUserId()
+        {
+            return UserId;
+        }
+
+        public UserAttributes GetAttributes()
+        {
+            return new UserAttributes(Attributes);
```
Comment (jaeopt) [inline OptimizelySDK/OptimizelyUserContext.cs:61]: Don't we need a mutex protection for this copy with the same lock for setAttribute?

**Review by jaeopt — CHANGES_REQUESTED** [review]
Missing a mutex lock while making a copy. Other than that, all looks good!

File: OptimizelySDK/OptimizelyUserContext.cs (hunk 1 above)
Comment (msohailhussain) [inline OptimizelySDK/OptimizelyUserContext.cs:61]: yes, mutex needed.

**Review by msohailhussain — APPROVED** [review]
lgtm

**Review by jaeopt — APPROVED** [review]
LGTM

**Pull request merged**